src/Operators/TensorOperators.jl

```diff
@@ -0,0 +1,126 @@
+# Produces a matrix representation of a tensor product of operators, C = A⊗B.
+# Like built-in `kron` but with permutation. Returns C_{acbd} = A_{ab} B_{cd}.
+function kron_operator(A::AbstractMatrix, B::AbstractMatrix)
```

Comment:
Is this one of the kron-like matrix products defined by `LinearMaps.jl`? [link](https://julialinearalgebra.github.io/LinearMaps.jl/stable/types/#Base.kron-Tuple{LinearMap,%20LinearMap}). It may be worth investigating whether `LinearMaps` helps in anyway with these onsite operators, [c.f.](https://github.com/SunnySuite/Sunny.jl/blob/148a051b39dca3e880a39b9ea3d76fe12bfea001/src/StructureFactors/StructureFactors.jl#L122). Seems unlikely because they are always dense matrices.

Also, is it faster than `Base.permutedims` + `Base.kron`?

Reply:
Good catch. The kron from LinearMaps is semantically the same. It might be lazy, which would save work at construction time, but I suspect it wouldn't significantly reduce the total amount of work required (e.g. there will eventually be an SVD operation).

Eventually we should have a chat about the cost of constructing Hamiltonians in the context of inverse modeling. If optimization and LSWT becomes really fast, it could be that building the Hamiltonian is the dominant cost (e.g. symmetry analysis can sometimes be really slow). I have some thoughts to run by you.